editor/editor_file_system.cpp

```diff
@@ -899,6 +942,9 @@ void EditorFileSystem::_scan_new_dir(EditorFileSystemDirectory *p_dir, Ref<DirAc
 					fi->type = "TextFile";
 				}
 				fi->uid = ResourceLoader::get_resource_uid(path);
+				if (fi->type == Resource::get_class_static()) {
```

Comment:
`type` can be any C++ class, so it might be more suitable to use an inheritance check here, no? We want to record the script path even if the script is attached to something more derived than the base `Resource` type.